### PR TITLE
refactor delay_before after checking state

### PR DESCRIPTION
**Pull request by AlfinST**
# Summary

@yahoo/ychaos-dev

refactored code so that framework sleeps only once state matches with the testplan

---

**Fixes**: #60 

#### Prerequisites
- [x] I have read the contribution guidelines

#### Code Analysis
- [x] Covered by Unittests
- [ ] Security warnings ignored (Why?!)

#### Project related
- [ ] Schema changed and is backward compatible.
- [ ] Introduces a new sub-command for ychaos CLI

### Autogenerated Files
- [ ] Auto generated schema

### Pre-Merge Checklist

- [ ] All the build jobs are passing

---

### Checklist (Reviewer 1)

- [ ] Reviewed Source Code
- [ ] Reviewed Tests (If applicable)
- [ ] Reviewed Documentation (If applicable)

### Checklist (Reviewer 2)

- [ ] Reviewed Source Code
- [ ] Reviewed Tests (If applicable)
- [ ] Reviewed Documentation (If applicable)

### CODE REVIEW

File: src/ychaos/core/verification/controller.py [hunk 1]
```diff
@@ -170,6 +167,9 @@ def execute(self) -> bool:
 
                 plugin = plugin_class(verification_plugin.config, data)
 
+                # Delay before verifying
+                time.sleep(verification_plugin.delay_before)
```
Comment (shashankrnr32) [inline src/ychaos/core/verification/controller.py:171]: Please add a mockito `verify` in UT to verify this

Comment (shashankrnr32) [reply src/ychaos/core/verification/controller.py:171]: time.sleep should not be called even once for the case

Comment (AlfinST) [reply src/ychaos/core/verification/controller.py:171]: right should'nt sleep for `delay_before `
wouldn't it be called once for `delay_after`
so the check would be to see if it only sleeps once?

Comment (shashankrnr32) [reply src/ychaos/core/verification/controller.py]: Yes

File: tests/core/verification/test_controller.py [hunk 2]
```diff
@@ -120,3 +122,17 @@ def __call__(self, *args):
...
+    def test_verification_controller_to_not_sleep_for_delay_before_if_in_different_system_state(
+        self,
+    ):
+
+        when(time).sleep(0).thenReturn(None)
+        verification_controller = VerificationController(
+            self.mock_testplan, SystemState.RECOVERED, list()
+        )
+        verification_controller.execute()
+        verify(time, times=1).sleep(0)
```
Comment (shashankrnr32) [inline tests/core/verification/test_controller.py:135]: Shouldnt this be 0 for not sleep?

Comment (AlfinST) [reply tests/core/verification/test_controller.py:135]: `sleep(delay_after)`
gets called regardless of system state

Comment (AlfinST) [reply tests/core/verification/test_controller.py:135]: was that also be refactored to sleep only on system state match?

Comment (shashankrnr32) [reply tests/core/verification/test_controller.py:135]: Can you then add verification it gets twice for the right state?

**Review by shashankrnr32 — APPROVED** [review]
Looks good 👍

**Review by saravanan-murugesan — APPROVED** [review]

**Comment by shashankrnr32** [comment]
Is this ready to be merged? @AlfinST

**Pull request merged**